### PR TITLE
Maintain global stake

### PR DESCRIPTION
**Pull request by mzhu25**
## Description
Adds a state variable to keep track of total stake in the system (by status), which gets updated every time `stake`, `unstake`, or `moveStake` is called. Intended for use in future features. 
Also makes some functions in `staker_actor` private
## Testing instructions
Updated `staker_actor.ts` to check global stake

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/staking/contracts/src/immutable/MixinStorage.sol [hunk 1]
```diff
@@ -47,6 +47,10 @@ contract MixinStorage is
     // address for read-only proxy to call
     address internal readOnlyProxyCallee;
 
+    // mapping from StakeStatus to the total amount of stake in that status for the entire
+    // staking system.
+    mapping (uint8 => uint256) internal globalStakeByStatus;
```
[the remark this answers is not in the document]
Comment (mzhu25) [reply contracts/staking/contracts/src/immutable/MixinStorage.sol]: you can't use enums for mapping keys 💀

[the remark this answers is not in the document]
Comment (jalextowle) [reply contracts/staking/contracts/src/immutable/MixinStorage.sol]: I sort of figured that's why you did that. Big RIP

File: contracts/staking/contracts/src/stake/MixinStakeBalances.sol [hunk 2]
```diff
@@ -53,6 +53,17 @@ contract MixinStakeBalances is
...
 
+    /// @dev Returns the total stake in a given status for the entire staking system.
+    /// @param status of stake being queried.
+    /// @return Total amount of stake in given status.
+    function getGlobalStakeInStatus(IStructs.StakeStatus status)
+        public
+        view
+        returns (uint256)
+    {
+        return globalStakeByStatus[uint8(status)];
```
Comment (abandeali1) [inline contracts/staking/contracts/src/stake/MixinStakeBalances.sol]: I feel like we can just remove this function and make `globalStakeByStatus` public (I'm going to be making that change throughout the staking contracts).

File: contracts/staking/contracts/src/immutable/MixinStorage.sol [hunk 3]
```diff
@@ -47,6 +47,10 @@ contract MixinStorage is
     // address for read-only proxy to call
     address internal readOnlyProxyCallee;
 
+    // mapping from StakeStatus to the total amount of stake in that status for the entire
+    // staking system.
+    mapping (uint8 => uint256) public globalStakeByStatus;
```
Comment (hysz) [inline contracts/staking/contracts/src/immutable/MixinStorage.sol]: We'll need to add this to `TestStorageLayout`

**Comment by mzhu25** [comment]
ended up refactoring `StakerActor` a bit

File: contracts/staking/contracts/src/immutable/MixinStorage.sol (hunk 3 above)
Comment (mzhu25) [reply contracts/staking/contracts/src/immutable/MixinStorage.sol]: `TestStorageLayout` is officially ded as of [2168](https://github.com/0xProject/0x-monorepo/pull/2168)

File: contracts/staking/test/actors/staker_actor.ts [hunk 4]
```diff
@@ -226,106 +243,96 @@ export class StakerActor extends BaseActor {
         expect(balances.delegatedStakeBalance.nextEpochBalance, 'delegated stake balance (next)').to.be.bignumber.equal(
             expectedBalances.delegatedStakeBalance.nextEpochBalance,
         );
+        expect(
```
Comment (hysz) [inline contracts/staking/test/actors/staker_actor.ts]: Nice 👍

**Pull request merged**